### ISSUE TITLE
Setting Genre Metadata Causes libsndfile error.

### ISSUE THREAD

**Issue by MaphsterB** [vokimon/python-wavefile#18]
Attempting to set the genre metadata field on a loaded wavefile will cause the following error to print from libsndfile:
`('genre', 54, 'Error : Bad string data type.')`

The issue lies with the way libsndfile defines constants, and how `__setattr__` works in the `WaveMetadata` class

```python
    SF_STR_TITLE       = 0x01
    SF_STR_COPYRIGHT   = 0x02
    SF_STR_SOFTWARE    = 0x03
    SF_STR_ARTIST      = 0x04
    SF_STR_COMMENT     = 0x05
    SF_STR_DATE        = 0x06
    SF_STR_ALBUM       = 0x07
    SF_STR_LICENSE     = 0x08
    SF_STR_TRACKNUMBER = 0x09
    SF_STR_GENRE       = 0x10
```

```python
    def __setattr__(self, name, value) :
        if name not in self.strings :
            return object.__setattr__(self, name, value)

        stringid = self.strings.index(name)+1
        error = _lib.sf_set_string(self._sndfile, stringid, value.encode(_tagencoding))
        if error : print(ValueError(
            name,
            error, _sferrormessage(error)))
```

When called with `name="genre"` this will look up the index of genre in the `strings` list and add 1 (resulting in 10). But 10 != 0x10, so an incorrect value is passed to `sf_set_string()`.

**Comment by lucas993** [comment]
:+1:

**Comment by vokimon** [comment]
Great finding! I really missed the gap because of the hex coding!!

The error went unnoticed by the tests because some strings (tracknumber and genre) were not available for ogg/vorbis for libsndfile-1.0.25, so they were commented out of the tests.

Working on a different approach to get the string ids and enabling the test of those strings for newer versions of libsndfile.
